docs/dev_tools/adding_exporters.md

```diff
@@ -1,74 +1,97 @@
+ * `resources` a `Resources` object that contains many lists of files that an exporter will find useful, like C and Cpp sources or header search paths.
 
-The setup code takes care of scanning for the resources used in the export process and collecting the configuration required to build the project at hand.
-These steps construct an object of one of the exporter plugin classes listed in the exporter map and populate the object with useful attributes.
-The attributes include:
- * `toolchain` a mbedToolchain object that may be used to query the configuration of the toolchain
-   * `toolchain.target` a Target object that may be use to query target configuration
- * `project_name` the name of the project
- * `flags` the flags that the mbedToolchain instance would use to compile the c/cpp/asm files if invoked
- * `resources` a Resources object that contains many lists of files that an exporter would find useful, like C and Cpp sources or header search paths
+___Plugin tools___
 
-The other half of the common code is a library for use by a plugin.
-This API includes:
- * `gen_file` use Jinja2 to generate a file from a template
- * `get_source_files` get all asembly, C, C++, and so on files as one list
- * `group_project_files` Group all files passed in into groups sutable for an IDE
+The other half of the common code is a library for use by a plugin. This API includes:
+
+ * `gen_file` use Jinja2 to generate a file from a template.
+ * `get_source_files` get all files for assembly, C, C++, and so on as one list.
+ * `group_project_files` group all files passed in. The groups will be suitable for an IDE.
 
 ### Plugin code
 
-Plugin code is contained within a sub-directory of the `tools/export` directory named after the IDE or Toolchain that the plugin is exporting for.
-For example, the Uvision exporter's templates and python code is contained within the directory `tools/export/uvision` and the Makefile exporter's code and templates within `tools/export/makefile`.
-The python code for the Plugin should be placed into an `__init__.py` file and should be imported into `tools/export/__init__.py` and added to the exporter map to be enabled.
+Plugin code is contained within a sub-directory of the `tools/export` directory named after the IDE or toolchain that the plugin is exporting for.
+For example, the uVision exporter's templates and Python code is contained within the directory `tools/export/uvision` and the Makefile exporter's code and templates within `tools/export/makefile`.
+
+The Python code for the plugin should be:
+
+1. Placed into an `__init__.py` file.
+1. Imported into `tools/export/__init__.py`.
+1. Added to the exporter map.
+
+___The `generate` method___
+
+Each exporter is expected to implement one method, `generate`, which is responsible for creating all of the required project files for the IDE or toolchain that the plugin targets. The method should provide the exporter's supported targets in a `TARGETS` class variable.
```

Comment:
The `generate` method does not provide the supported targets. It's run on an object, where as `TARGETS` is a class variable.